ChangeLog.d/bugfix_3794.txt

```diff
@@ -0,0 +1,3 @@
+Bugfix
+  * Fix handling of EOF against 0xff bytes and on platforms with
+    unsigned chars. Fixes #3794.
```

Comment:
Please state explicitly that this removes a build failure on platforms where char is unsigned, both in the changelog entry and in the commit message.